packages/core/src/StateNode.ts

```diff
@@ -476,9 +476,10 @@ class StateNode<
       id: this.id,
       key: this.key,
       version: this.version,
+      context: this.context!,
       type: this.type,
       initial: this.initial,
-      history: this.history,
+      history: this.history || undefined,
```

Comment:
can history be some other falsy value that `undefined` fallback is needed here?

Reply:
Yes - removing this will make validation fail.

Reply:
Right - but this is an outcome. I'm wondering about the root cause 😃 Maybe this can be avoided by handling things better "upstream"?